### ISSUE TITLE
WET LeftMenu doesn't set the "wb-navcurr" on current page's element.

### ISSUE THREAD

**Issue by joel-ld-esdc** [wet-boew/wet-boew#8655]
### Problem Found in
**Version:*** 1.30.0 (4.0.30)  
**Template:*** MVC, WebForm  
**Environment:***  *AKAMAI  
**Master/Layout:** (all).LeftMenu   
**Additional settings:** 

### Problem Description:

Me and my colleagues saw that the left menu doesn't work as expected since the last version of the GoC template.
The left menu's item associated to the current page should be "selected/highlighted". But it doesn't.

### Expected result
![image](https://user-images.githubusercontent.com/34245447/58804181-3c74ce80-85df-11e9-9be4-db6df5ba11ab.png)

### Current result
![image](https://user-images.githubusercontent.com/34245447/58804194-41d21900-85df-11e9-80b6-14e48b2d2bfc.png)

### Analysis 
The class "wb-navcurr" is missing on the <a> element.

I also take a look into the wet-boew.js file and the logic around the left menu looks good. There is no changes.

Any issues ?

**Comment by shawnthompson** [comment]
I've tested it in the latest GCWeb dist and I can confirm it's a problem there also. It worked with version 4.0.29

**Comment by joel-ld-esdc** [comment]
Also, I take a look to the wet-boew.js on the function that generate the wb-navcurr class and everything looks fine. There is no changes since older release.

**Comment by duboisp** [comment]
There were markup change with [GCWeb v5 as documented](https://wet-boew.github.io/themes-dist/GCWeb/gcweb-theme/v5-migration.html).

@joel-ld-esdc , @shawnthompson are you experience that issue with a standalone version of the latest GCWeb?

**Comment by joel-ld-esdc** [comment]
@duboisp , I'm getting this error into our existing application using the DotNetTemplate from SDS.

**Comment by duboisp** [comment]
ok, we will just need to replicate your experienced error with a standalone version of GCWeb.

**Comment by thekodester** [comment]
Using the example here https://wet-boew.github.io/v4.0-ci/theme/content-secmenu-en.html and the class `wb-navcurr` it works.

**Comment by joel-ld-esdc** [comment]
@thekodester Exact, If I set the class manually into the sidemenu, it works. 

But for unknown reason, the function is not triggered.

**Comment by thekodester** [comment]
@joel-ld-esdc Sorry what I was trying to say is the secondary menu has no logic and must be set manually. Also the navcurrent plugin doesn't look at the secondary menu unless you add `wb-navcurr` to the `nav` element, but in my testing even adding `wb-navcurr` it seems to be ignoring it.

**Comment by joel-ld-esdc** [comment]
@thekodester 
yes, I saw this plugin and it was present both in old and new version.

In past version, the plugin was triggered i think because the `wb-navcurr` was present for the current page.

Since the 4.0.30, the class is not present anymore.

**Comment by namjohn920** [comment]
@duboisp 
![toggle_focused_before](https://user-images.githubusercontent.com/40031678/71363673-e447ee80-2567-11ea-8824-4bc2a0a16497.png)

so basically what the issue is that without adding wb-curr class to the list items on the secondary menu, the menu does not stay focused. (if I click the menu item and click something else, the focus goes away)

![toggle_focused_after](https://user-images.githubusercontent.com/40031678/71363739-19544100-2568-11ea-9fbb-45ae8bc895d2.png)
as shown in the picture, once wb-curr class is added to each item, and it works. 

the file I used to replicate the issue can be found at /theme/content-secmenu-en.html

**Comment by namjohn920** [comment]
@duboisp 
![image](https://user-images.githubusercontent.com/40031678/71370710-bae58d80-257c-11ea-9ac6-5ce7ad0502fa.png)
As soon as the link in the litem (href link) matches with the current address of the page, it works. 
It did not stay focused due to missing page link.

**Comment by EricDunsworth** [comment]
Greetings from the future :)...

This probably doesn't really matter at this point, but I believe I know what caused this.

While testing updates for #9307 in the [WET theme's secondary menu demo page](https://wet-boew.github.io/v4.0-ci/theme/content-secmenu-en.html), I noticed that the [navcurrent plugin](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/navcurrent/navcurrent.js) still runs even after removing all hardcoded references to the ``wb-navcurr`` class...

What seems to be happening is that the [WET's menu plugin](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/menu/menu.js) automatically triggers the [navcurrent event](https://github.com/wet-boew/wet-boew/blob/98eda2aded8ca4d5bbb19140d775b0169f230309/src/plugins/menu/menu.js#L20) on these:
* [Mega menu](https://github.com/wet-boew/wet-boew/blob/98eda2aded8ca4d5bbb19140d775b0169f230309/src/plugins/menu/menu.js#L350) (``.wb-menu`` class)
* [Mobile menu](https://github.com/wet-boew/wet-boew/blob/98eda2aded8ca4d5bbb19140d775b0169f230309/src/plugins/menu/menu.js#L351) (``sm-pnl`` ID)
* [Left navigation menu](https://github.com/wet-boew/wet-boew/blob/98eda2aded8ca4d5bbb19140d775b0169f230309/src/plugins/menu/menu.js#L276) (``#wb-sec`` ID)
* [Footer](https://github.com/wet-boew/wet-boew/blob/98eda2aded8ca4d5bbb19140d775b0169f230309/src/plugins/menu/menu.js#L306) (``#wb-info`` ID)

So... what happened with GCWeb? [Its mega menu](https://web.archive.org/web/20190923032558/https://wet-boew.github.io/themes-dist/GCWeb/localnav/task1/index-en.html) was replaced with a ["MENU" dropdown](https://web.archive.org/web/20190923032558/https://wet-boew.github.io/themes-dist/GCWeb/localnav/task1/index-en.html) in 2019. As a result, pretty much all of its page templates lost out on the menu plugin's "automatic" navcurrent implementation. Ditto for Canada.ca.

As for whether it'd be worth trying to address it in 2026... IMO probably not. It's unlikely many (any?) pages are still using left navigation menus on Canada.ca anymore. Plus GCWeb seems to be gradually moving towards the [GC Design System (GCDS)](https://design-system.canada.ca/) these days.
